js_modules/dagster-ui/packages/ui-core/src/assets/RecentUpdatesTimeline.tsx

```diff
@@ -29,67 +29,69 @@ const INNER_TICK_WIDTH = 4;
 const MIN_TICK_WIDTH = 5;
 const BUCKETS = 50;
 
-type MaterializationType = ReturnType<typeof useRecentAssetEvents>['materializations'][0];
-type ObservationType = ReturnType<typeof useRecentAssetEvents>['observations'][0];
+type AssetEventType = ReturnType<typeof useRecentAssetEvents>['events'][0];
 type Props = {
   assetKey: AssetKey;
-  materializations?: MaterializationType[];
-  observations?: ObservationType[];
+  events?: AssetEventType[];
   loading: boolean;
 };
 
 export const RecentUpdatesTimelineForAssetKey = memo((props: {assetKey: AssetKey}) => {
-  const data = useRecentAssetEvents(
-    props.assetKey,
-    100,
-    MaterializationHistoryEventTypeSelector.ALL,
+  const data = useRecentAssetEvents(props.assetKey, 100, [
+    AssetEventHistoryEventTypeSelector.MATERIALIZATION,
+    AssetEventHistoryEventTypeSelector.FAILED_TO_MATERIALIZE,
+    AssetEventHistoryEventTypeSelector.OBSERVATION,
+  ]);
+  return (
+    <RecentUpdatesTimeline assetKey={props.assetKey} events={data.events} loading={data.loading} />
   );
-  return <RecentUpdatesTimeline assetKey={props.assetKey} {...data} />;
 });
 
-export const RecentUpdatesTimeline = ({
-  assetKey,
-  observations,
-  materializations,
-  loading,
-}: Props) => {
+export const RecentUpdatesTimeline = ({assetKey, events, loading}: Props) => {
   const {containerProps, viewport} = useViewport();
   const widthAvailablePerTick = viewport.width / BUCKETS;
 
   const tickWidth = Math.max(widthAvailablePerTick, MIN_TICK_WIDTH);
 
   const buckets = Math.floor(viewport.width / tickWidth);
 
-  const sortedMaterializations = useMemo(() => {
-    if (!materializations && observations) {
-      const seenTimestamps = new Set();
-      return observations
-        .map((obs) => {
-          const lastUpdated = obs.metadataEntries.find(
+  const enrichedEvents = useMemo(() => {
+    const seenTimestamps = new Set();
+    return events
+      ?.map((event) => {
+        if (
+          event.__typename === 'MaterializationEvent' ||
+          event.__typename === 'FailedToMaterializeEvent'
+        ) {
+          return event;
+        } else {
+          const lastUpdated = event.metadataEntries.find(
```

Comment:
I was getting errors when i ran locally because of this code section. Some observation events i emitted didn't have this `TimestampMetadataEntry` entry field, and therefore `lastUpdated` was undefined and the `.timestamp` failed. I feel like it makes sense to fall back on the event timestamp here if the metadata doesn't exist, but i'm not sure why this was working in the first place

Reply:
maybe it was because if any materializations existed we didn't show observations at all, so this code wasn't hit?